### DISCUSSION TITLE
Adding git sparse checkout support

### DISCUSSION THREAD

**Discussion by GraceDmello** [git-lfs/git-lfs#5686] [General] [answered]
Hi, I am planning to invest some time this week to add sparse checkout support. I am not sure where to start within the repo. Some direction or assistance would be helpful.

Thank you.

**Comment by bk2204** [comment, marked as answer]
My general approach to this would be to look through the existing issues and see what isn't supported and then try some basic testing with the repository.  For example, do `git lfs pull`, `git lfs checkout`, `git lfs push`, etc. work, and if they don't, what's needed to fix them?

Once you know what's not working, it should be pretty easy with `GIT_TRACE=1` to find out the failing Git calls, and you can then try to figure out how to fix them.  Note that we've had some discussion that at least in some cases, we'll need changes to Git (e.g., a sparse mode for `git ls-tree` or equivalent information in another mode) to properly support all the pieces in Git LFS, so that's a thing we'll need to consider.
